arcgis-ios-sdk-samples/Display information/Display grid/DisplayGridSettingsViewController.swift

```diff
@@ -241,77 +241,105 @@ class DisplayGridSettingsViewController: UITableViewController {
+            self.mapView?.grid?.labelPosition = AGSGridLabelPosition(rawValue: newIndex)!
+        }
+        optionsViewController.title = "Position"
+        show(optionsViewController, sender: self)
+    }
+    
+    private func showLabelFormatPicker() {
+        if let grid = mapView?.grid,
+            let selectedIndex = (grid as? AGSLatitudeLongitudeGrid)?.labelFormat.rawValue {
+            let optionsViewController = OptionsTableViewController(labels: labelFormatLabels, selectedIndex: selectedIndex) { (newIndex) in
+                if let grid = self.mapView?.grid as? AGSLatitudeLongitudeGrid {
+                    grid.labelFormat = AGSLatitudeLongitudeGridLabelFormat(rawValue: newIndex)!
+                    self.updateLabelFormatUI()
                 }
-                optionsViewController.title = "Grid Type"
-                show(optionsViewController, sender: self)
             }
-        case labelPositionCell:
-            let selectedIndex = grid.labelPosition.rawValue
-            let optionsViewController = OptionsTableViewController(labels: labelPositionLabels, selectedIndex: selectedIndex) { (newIndex) in
-                self.mapView?.grid?.labelPosition = AGSGridLabelPosition(rawValue: newIndex)!
-            }
-            optionsViewController.title = "Position"
+            optionsViewController.title = "Format"
             show(optionsViewController, sender: self)
-        case labelFormatCell:
-            if let selectedIndex = (grid as? AGSLatitudeLongitudeGrid)?.labelFormat.rawValue {
-                let optionsViewController = OptionsTableViewController(labels: labelFormatLabels, selectedIndex: selectedIndex) { (newIndex) in
-                    if let grid = self.mapView?.grid as? AGSLatitudeLongitudeGrid {
-                        grid.labelFormat = AGSLatitudeLongitudeGridLabelFormat(rawValue: newIndex)!
-                        self.updateLabelFormatUI()
-                    }
-                }
-                optionsViewController.title = "Format"
-                show(optionsViewController, sender: self)
-            }
-        case labelUnitCell:
-            if let selectedIndex = (grid as? AGSUSNGGrid)?.labelUnit.rawValue ?? (grid as? AGSMGRSGrid)?.labelUnit.rawValue {
-                let optionsViewController = OptionsTableViewController(labels: labelUnitLabels, selectedIndex: selectedIndex) { (newIndex) in
-                    if let grid = self.mapView?.grid as? AGSMGRSGrid {
-                        grid.labelUnit = AGSMGRSGridLabelUnit(rawValue: newIndex)!
-                    } else if let grid = self.mapView?.grid as? AGSUSNGGrid {
-                        grid.labelUnit = AGSUSNGGridLabelUnit(rawValue: newIndex)!
-                    }
-                    self.updateLabelUnitUI()
+        }
+    }
+    
+    private func showLabelUnitPicker() {
+        if let grid = mapView?.grid,
```

Comment:
Would this make more sense as a guard statement?

Reply:
👍